### PR TITLE
Enable Qwen1-5

### PR DESCRIPTION
**Pull request by intellinjun**
## Type of Change

feature or bug fix or documentation or others
API changed or not

## Description
[qwen1-5](https://huggingface.co/Qwen/Qwen1.5-7B-Chat) enabling
detail description 
- new convert function
- new forward function
- gguf format

### CODE REVIEW

File: neural_speed/convert/convert_qwen2.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#
+# Usage:
+#
+#   python3 models/convert-h5-to-ne.py
+#
+# This script is similar to "convert-pt-to-ne.py"
+#
+
+import io
+import os
```
Comment (a32543254) [inline neural_speed/convert/convert_qwen2.py:26]: could combine the convert_qwen and convert_qwen2 together ?

**Review by a32543254 — APPROVED** [review]
LGTM

File: neural_speed/models/qwen/qwen_utils.cpp [hunk 2]
```diff
@@ -150,6 +163,34 @@ void QWEN::load(model_context* ctx, model_progress_callback progress_callback, v
       layer.ffn[1] = ml->get_tensor(layers_i + ".mlp.w2.weight", {n_embd, n_ff}, backend);
       layer.ffn[2] = ml->get_tensor(layers_i + ".mlp.c_proj.weight", {n_ff, n_embd}, backend);
     }
+  } else {
```
Comment (Zhenzhong1) [inline neural_speed/models/qwen/qwen_utils.cpp:166]: There are so many if-elses, can we split those if-else and add comments to make it clear whether it is qwen1, qwen2, gguf or bin?

**Review by Zhenzhong1 — APPROVED** [review]

**Review by airMeng — APPROVED** [review]
An extension test is needed

File: neural_speed/convert/convert_qwen2.py (hunk 1 above)
Comment (intellinjun) [reply neural_speed/convert/convert_qwen2.py]: done

File: neural_speed/models/qwen/qwen_utils.cpp (hunk 2 above)
Comment (intellinjun) [reply neural_speed/models/qwen/qwen_utils.cpp]: done

**Review by zhentaoyu — APPROVED** [review]

File: neural_speed/models/qwen/qwen.cpp [hunk 3]
```diff
@@ -102,6 +102,12 @@ static bool qwen_model_eval_internal(model_context* ctx, const model_input* inpu
   const int n_vocab = hparams.n_vocab;
   const int n_rot = hparams.n_rot;
   const int head_dim = n_embd / n_head;
+  int qwen_version = 0;
```
[the remark this answers is not in the document]
Comment (intellinjun) [reply neural_speed/models/qwen/qwen.cpp:105]: ok

**Comment by intellinjun** [comment]
https://inteltf-jenk.sh.intel.com/job/neural_speed_extension/66/

**Comment by Zhenzhong1** [comment]
Please check this https://huggingface.co/Qwen/Qwen1.5-7B-Chat-GGUF and update supported_models.md

**Comment by a32543254** [comment]
please also add mixtral 8*7B in support model list

**Comment by intellinjun** [comment]
<html>
<body>
<div ccp_infra_version='3' ccp_infra_timestamp='1709602619369' ccp_infra_user_hash='1069609727' ccp_infra_copy_id='1518902b-e5a4-4c31-814d-ffa2d2f6c888' data-ccp-timestamp='1709602619369'><html><head><meta name=ProgId content=Excel.Sheet><meta name=Generator content="Microsoft Excel 15"><style>table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:12.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:等线, sans-serif;
	mso-font-charset:0;
	text-align:general;
	vertical-align:bottom;
	border:none;
	white-space:nowrap;
	mso-rotate:0;}
.xl101
	{font-size:11.0pt;
	text-align:center;
	vertical-align:middle;}
.xl148
	{color:#2A2A2A;
	font-size:11.0pt;
	font-family:Verdana, sans-serif;
	mso-font-charset:0;
	text-align:center;
	vertical-align:middle;
	border-top:1.0pt solid #CDCDCD;
	border-right:1.0pt solid #CDCDCD;
	border-bottom:none;
	border-left:1.0pt solid #CDCDCD;}
.xl220
	{color:black;
	font-size:11.0pt;
	font-family:宋体, sans-serif;
	mso-font-charset:0;}
</style></head><body link="#0563C1" vlink="#954F72">
Input | Output | Batchsize | Cores/Instance | Precision | Next Token(ms) | Memory mean used (Top 50%) MB | First Token(ms) | Total Latency(ms) | P90 Latency(ms) | P99 Latency(ms)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
32 | 32 | 1 | 32 | q4_j_i8_g128 | 27.15 | 4190.4 | 72.9 | 914.54 | 27.38 | 72.9
1024 | 32 | 1 | 32 | q4_j_i8_g128 | 29.77 | 4343.65 | 1041.29 | 1964.16 | 35.02 | 1041.29
2012 | 32 | 1 | 32 | q4_j_i8_g128 | 31.32 | 4712.84 | 2147.69 | 3118.68 | 31.95 | 2147.69
32 | 32 | 1 | 48 | q4_j_i8_g128 | 24.56 | 5532.91 | 57.06 | 818.42 | 24.73 | 57.06
1024 | 32 | 1 | 48 | q4_j_i8_g128 | 28.16 | 5279.89 | 859.34 | 1732.33 | 27.37 | 859.34
2012 | 32 | 1 | 48 | q4_j_i8_g128 | 29.84 | 5257.13 | 1776.02 | 2701.09 | 30.09 | 1776.02
32 | 32 | 1 | 56 | q4_j_i8_g128 | 25.31 | 5481.34 | 62.05 | 846.77 | 25.51 | 62.05
1024 | 32 | 1 | 56 | q4_j_i8_g128 | 27.56 | 5338.05 | 769.49 | 1623.89 | 27.81 | 769.49
2012 | 32 | 1 | 56 | q4_j_i8_g128 | 30.27 | 5314.82 | 1582.25 | 2520.77 | 30.45 | 1582.25
32 | 32 | 1 | 32 | q4_j_i8_g32 | 31.7 | 5164.92 | 190.52 | 1173.22 | 31.96 | 190.52
1024 | 32 | 1 | 32 | q4_j_i8_g32 | 36.67 | 4472.71 | 2072.26 | 3208.92 | 44.24 | 2072.26
2012 | 32 | 1 | 32 | q4_j_i8_g32 | 36.7 | 4930.47 | 4091.69 | 5229.29 | 36.9 | 4091.69
32 | 32 | 1 | 48 | q4_j_i8_g32 | 29.94 | 5442.93 | 307.86 | 1236.01 | 30.06 | 307.86
1024 | 32 | 1 | 48 | q4_j_i8_g32 | 31.97 | 5262.03 | 1756.79 | 2747.76 | 32.16 | 1756.79
2012 | 32 | 1 | 48 | q4_j_i8_g32 | 35.19 | 5311.04 | 3680.12 | 4771.07 | 35.39 | 3680.12
32 | 32 | 1 | 56 | q4_j_i8_g32 | 30.69 | 5432.81 | 160.35 | 1111.7 | 30.89 | 160.35
1024 | 32 | 1 | 56 | q4_j_i8_g32 | 32.89 | 5323.78 | 1641.17 | 2660.72 | 33.09 | 1641.17
2012 | 32 | 1 | 56 | q4_j_i8_g32 | 36.06 | 5347.23 | 3443.91 | 4561.86 | 36.54 | 3443.91
32 | 32 | 1 | 32 | q4_0 | 34.39 | 4796.24 | 457.98 | 1524.17 | 39.44 | 457.98
1024 | 32 | 1 | 32 | q4_0 | 33.11 | 4762.58 | 12449.43 | 13475.96 | 33.43 | 12449.43
2012 | 32 | 1 | 32 | q4_0 | 35.47 | 5655.83 | 26196.5 | 27296.14 | 35.66 | 26196.5
32 | 32 | 1 | 48 | q4_0 | 27.01 | 5344.82 | 329.98 | 1167.39 | 27.25 | 329.98
1024 | 32 | 1 | 48 | q4_0 | 30.27 | 5277.15 | 9383.2 | 10321.59 | 30.46 | 9383.2
2012 | 32 | 1 | 48 | q4_0 | 32.89 | 5542.24 | 19753.62 | 20773.2 | 33.06 | 19753.62
32 | 32 | 1 | 56 | q4_0 | 27.48 | 5377.97 | 314.21 | 1166.22 | 27.74 | 314.21
1024 | 32 | 1 | 56 | q4_0 | 30.22 | 5327.12 | 8735.25 | 9672.2 | 30.58 | 8735.25
2012 | 32 | 1 | 56 | q4_0 | 32.63 | 5489.69 | 18234.3 | 19245.84 | 32.87 | 18234.3

</body></html></div>
</body>
</html>

**Comment by intellinjun** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/linjun1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/linjun1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link=blue vlink=purple>


qwen-1_5 | 32 | 32 | 1 | 32 | q4_j_i8_g128 |   | New | 27.15 | 4190.4 | 72.9 | 914.54 | 27.38 | 72.9
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
qwen-1_5 | 1024 | 32 | 1 | 32 | q4_j_i8_g128 |   | New | 29.77 | 4343.65 | 1041.29 | 1964.16 | 35.02 | 1041.29
qwen-1_5 | 2012 | 32 | 1 | 32 | q4_j_i8_g128 |   | New | 31.32 | 4712.84 | 2147.69 | 3118.68 | 31.95 | 2147.69
qwen-1_5 | 32 | 32 | 1 | 48 | q4_j_i8_g128 |   | New | 24.56 | 5532.91 | 57.06 | 818.42 | 24.73 | 57.06
qwen-1_5 | 1024 | 32 | 1 | 48 | q4_j_i8_g128 |   | New | 28.16 | 5279.89 | 859.34 | 1732.33 | 27.37 | 859.34
qwen-1_5 | 2012 | 32 | 1 | 48 | q4_j_i8_g128 |   | New | 29.84 | 5257.13 | 1776.02 | 2701.09 | 30.09 | 1776.02
qwen-1_5 | 32 | 32 | 1 | 56 | q4_j_i8_g128 |   | New | 25.31 | 5481.34 | 62.05 | 846.77 | 25.51 | 62.05
qwen-1_5 | 1024 | 32 | 1 | 56 | q4_j_i8_g128 |   | New | 27.56 | 5338.05 | 769.49 | 1623.89 | 27.81 | 769.49
qwen-1_5 | 2012 | 32 | 1 | 56 | q4_j_i8_g128 |   | New | 30.27 | 5314.82 | 1582.25 | 2520.77 | 30.45 | 1582.25
qwen-1_5 | 32 | 32 | 1 | 32 | q4_j_i8_g32 |   | New | 31.7 | 5164.92 | 190.52 | 1173.22 | 31.96 | 190.52
qwen-1_5 | 1024 | 32 | 1 | 32 | q4_j_i8_g32 |   | New | 36.67 | 4472.71 | 2072.26 | 3208.92 | 44.24 | 2072.26
qwen-1_5 | 2012 | 32 | 1 | 32 | q4_j_i8_g32 |   | New | 36.7 | 4930.47 | 4091.69 | 5229.29 | 36.9 | 4091.69
qwen-1_5 | 32 | 32 | 1 | 48 | q4_j_i8_g32 |   | New | 29.94 | 5442.93 | 307.86 | 1236.01 | 30.06 | 307.86
qwen-1_5 | 1024 | 32 | 1 | 48 | q4_j_i8_g32 |   | New | 31.97 | 5262.03 | 1756.79 | 2747.76 | 32.16 | 1756.79
qwen-1_5 | 2012 | 32 | 1 | 48 | q4_j_i8_g32 |   | New | 35.19 | 5311.04 | 3680.12 | 4771.07 | 35.39 | 3680.12
qwen-1_5 | 32 | 32 | 1 | 56 | q4_j_i8_g32 |   | New | 30.69 | 5432.81 | 160.35 | 1111.7 | 30.89 | 160.35
qwen-1_5 | 1024 | 32 | 1 | 56 | q4_j_i8_g32 |   | New | 32.89 | 5323.78 | 1641.17 | 2660.72 | 33.09 | 1641.17
qwen-1_5 | 2012 | 32 | 1 | 56 | q4_j_i8_g32 |   | New | 36.06 | 5347.23 | 3443.91 | 4561.86 | 36.54 | 3443.91
qwen-1_5 | 32 | 32 | 1 | 32 | q4_0 |   | New | 34.39 | 4796.24 | 457.98 | 1524.17 | 39.44 | 457.98
qwen-1_5 | 1024 | 32 | 1 | 32 | q4_0 |   | New | 33.11 | 4762.58 | 12449.43 | 13475.96 | 33.43 | 12449.43
qwen-1_5 | 2012 | 32 | 1 | 32 | q4_0 |   | New | 35.47 | 5655.83 | 26196.5 | 27296.14 | 35.66 | 26196.5
qwen-1_5 | 32 | 32 | 1 | 48 | q4_0 |   | New | 27.01 | 5344.82 | 329.98 | 1167.39 | 27.25 | 329.98
qwen-1_5 | 1024 | 32 | 1 | 48 | q4_0 |   | New | 30.27 | 5277.15 | 9383.2 | 10321.59 | 30.46 | 9383.2
qwen-1_5 | 2012 | 32 | 1 | 48 | q4_0 |   | New | 32.89 | 5542.24 | 19753.62 | 20773.2 | 33.06 | 19753.62
qwen-1_5 | 32 | 32 | 1 | 56 | q4_0 |   | New | 27.48 | 5377.97 | 314.21 | 1166.22 | 27.74 | 314.21
qwen-1_5 | 1024 | 32 | 1 | 56 | q4_0 |   | New | 30.22 | 5327.12 | 8735.25 | 9672.2 | 30.58 | 8735.25
qwen-1_5 | 2012 | 32 | 1 | 56 | q4_0 |   | New | 32.63 | 5489.69 | 18234.3 | 19245.84 | 32.87 | 18234.3



</body>

</html>

**Pull request merged**